### PR TITLE
nonmaxsupression Layer

### PR DESCRIPTION
**Pull request by LakshmiKumar23**
1. layer created
2. model compiler changes
3. changed utilities/vxTensor.cpp to read int64 type tensor

### CODE REVIEW

**Comment by kiritigowda** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5789000&bid=18950399)

File: model_compiler/python/nnir.py [hunk 1]
```diff
@@ -731,13 +737,22 @@ def updateLocals(self):
                     local.setInfo(output_type, output_shape)
                     local.setFormat(input.format)
                     self.addLocal(local)
+                elif node.type in ['nms']:
+                    output_type = 'I064'
+                    output_shape = [1,1,,3]
```
Comment (kiritigowda) [inline model_compiler/python/nnir.py:742]: Codacy found an issue: [invalid syntax (F999)](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5789000)

File: amd_openvx_extensions/amd_nn/src/nms_layer.cpp [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+                    float overlap;
+                    if (center_point_box == 0) /*indicates box data = [y1,x1,y2,x2] - mostly TF models */
+                    {
+                        overlap = computeOverlapCoordinates(boxes[idx], boxes[prev_idx]);
+                    }
+                    else if(center_point_box == 1) /*indicates box data = [x_center,y_center,width,height] - mostly PyTorch models*/
+                    {
+                        overlap = computeOverlapCenter(boxes[idx], boxes[prev_idx]);
+                    }
+                    if(overlap <= iou_thresh[0])
```
Comment (kiritigowda) [inline amd_openvx_extensions/amd_nn/src/nms_layer.cpp:404]: Codacy found an issue: [Uninitialized variable: overlap](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5789000)

**Comment by kiritigowda** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- model_compiler/python/nnir.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5789000&bid=18950399)

File: model_compiler/python/nnir.py [hunk 3]
```diff
@@ -731,13 +737,22 @@ def updateLocals(self):
                     local.setInfo(output_type, output_shape)
                     local.setFormat(input.format)
                     self.addLocal(local)
+                elif node.type in ['nms']:
+                    output_type = 'I064'
+                    output_shape = [1,1,1,3]
```
Comment (kiritigowda) [inline model_compiler/python/nnir.py:742]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5789000)

File: amd_openvx_extensions/amd_nn/src/nms_layer.cpp (hunk 2 above)
Comment (kiritigowda) [inline amd_openvx_extensions/amd_nn/src/nms_layer.cpp:404]: Codacy found an issue: [Uninitialized variable: overlap](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5789000)

**Comment by kiritigowda** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- model_compiler/python/nnir.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5789000&bid=18950399)

File: model_compiler/python/nnir.py (hunk 3 above)
Comment (kiritigowda) [inline model_compiler/python/nnir.py:742]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5789000)

File: amd_openvx_extensions/amd_nn/src/nms_layer.cpp (hunk 2 above)
Comment (kiritigowda) [inline amd_openvx_extensions/amd_nn/src/nms_layer.cpp:404]: Codacy found an issue: [Uninitialized variable: overlap](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5789000)

**Comment by hansely** [comment]
Tested with gdfs and the output matches with the onnx examples. @LakshmiKumar23 @kiritigowda

**Comment by kiritigowda** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- model_compiler/python/nnir.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5789000&bid=18950399)

File: amd_openvx_extensions/amd_nn/src/nms_layer.cpp (hunk 2 above)
Comment (kiritigowda) [inline amd_openvx_extensions/amd_nn/src/nms_layer.cpp:404]: Codacy found an issue: [Uninitialized variable: overlap](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5789000)

**Comment by kiritigowda** [comment]
@LakshmiKumar23 set overlap to default value as a fail safe. Fails tests - - if(overlap <= iou_thresh[0])

**Comment by kiritigowda** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- model_compiler/python/nnir.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5789000&bid=18950399)

**Comment by kiritigowda** [comment]
@LakshmiKumar23 can you sync with TOT to resolve conflicts.

**Comment by LakshmiKumar23** [comment]
@kiritigowda resolved merge conflicts

**Comment by kiritigowda** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- model_compiler/python/nnir.py  1
         

Clones added
============
- amd_openvx_extensions/amd_nn/src/nms_layer.cpp  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5789000&bid=18950399)

**Pull request merged**